File: src/main/java/com/m2t/hachikodictionary/dto/word/UpdateLearnedWordRequest.java
package com.m2t.hachikodictionary.dto.word;

import jakarta.validation.constraints.NotBlank;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UpdateLearnedWordRequest {
    @NotBlank
    private String accountId;
    @NotBlank
    private String wordId;
    private Boolean result;
}

File: src/main/java/com/m2t/hachikodictionary/dto/word/UpdateLearnedWordRequest.kt
package com.m2t.hachikodictionary.dto.word

import jakarta.validation.constraints.NotBlank
import org.springframework.lang.NonNull

data class UpdateLearnedWordRequest(
    @field:NotBlank(message = "accountId must not be blank")
    val accountId: String,
    @field:NotBlank(message = "wordId must not be blank")
    val wordId: String,
    val result: Boolean
) {
}